pkg/dockerregistry/client.go

```diff
@@ -597,6 +603,87 @@ func (repo *v2repository) getImage(c *connection, image, userTag string) (*Image
 	return repo.getTaggedImage(c, image, userTag)
 }
 
+func (repo *v2repository) getImageConfig(c *connection, dgst string) ([]byte, error) {
+	endpoint := repo.endpoint
+	endpoint.Path = path.Join(endpoint.Path, fmt.Sprintf("/v2/%s/blobs/%s", repo.name, dgst))
+	req, err := http.NewRequest("GET", endpoint.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %v", err)
+	}
+
+	if len(repo.token) > 0 {
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", repo.token))
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, convertConnectionError(c.url.String(), fmt.Errorf("error getting image config for %s: %v", repo.name, err))
+	}
+	defer resp.Body.Close()
+
+	switch code := resp.StatusCode; {
+	case code == http.StatusUnauthorized:
+		if len(repo.token) != 0 {
+			// The DockerHub returns JWT tokens that take effect at "now" at second resolution, which means clients can
+			// be rejected when requests are made near the time boundary.
+			if repo.retries > 0 {
```

Comment:
Actually, we're using this pattern in the registry already several times. I suggest leaving this as is.